spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -140,8 +140,19 @@ protected void doStart() {
 		if (isRunning()) {
 			return;
 		}
-		setRunning(true);
-		Object messageListener = getContainerProperties().getMessageListener();
+		ContainerProperties containerProperties = getContainerProperties();
+
+		if (!this.consumerFactory.isAutoCommit()) {
+			AckMode ackMode = containerProperties.getAckMode();
+			if (ackMode.equals(AckMode.BATCH) || ackMode.equals(AckMode.COUNT) || ackMode.equals(AckMode.COUNT_TIME)) {
+				Assert.state(containerProperties.getAckCount() > 0, "'ackCount' must be > 0");
```

Comment:
The other question is what should we do for MANUAL ? It is in theory the same as COUNT_TIME, but using manual acks instead of container-managed.  Maybe we can just document that MANUAL is identical with MANUAL_IMMEDIATE_SYNC if no count or time are set.